File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var checkIfExist = function (arr) {
    let chekMap = {};

    for (let num of arr) {
        if (chekMap[num << 1] || ((num & 1) === 0 && chekMap[num >> 1])) {
            return true;
        }
        chekMap[num] = true;
    }
    return false;
};

File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.go
func checkIfExist(arr []int) bool {
	m := make(map[int]struct{})

	for i := 0; i < len(arr); i++ {
		if arr[i] & 1 == 0 {
			if _, ok := m[arr[i]  >> 1]; ok {
				return true
			}
		}

		if _, ok := m[arr[i] << 1]; ok {
			return true
		}

		m[arr[i]] = struct{}{}
	}

	return false
}